libs/model/src/models/thread.ts

```diff
@@ -206,12 +206,15 @@ export default (
     models.Thread.belongsToMany(models.Address, {
       through: models.Collaboration,
       as: 'collaborators',
+      foreignKey: { name: 'thread_id', allowNull: false },
     });
     models.Thread.hasMany(models.Reaction, {
       foreignKey: 'thread_id',
       as: 'reactions',
     });
-    models.Thread.hasMany(models.Collaboration);
+    models.Thread.hasMany(models.Collaboration, {
+      foreignKey: { name: 'thread_id', allowNull: false },
```

Comment:
Why define `allowNull` in the association here?

Reply:
.sync will allow nulls in thread_id if we don't also flag it in the associations

Reply:
`allowNull: false` is already set on the `collaboration.ts` model. I tested removing `allowNull: false` from here and I get the same result in the DB. Are you 100% sure we need this? Would be nice if we could avoid defining column constraints in the associations.

Reply:
removed